static/js/notifications.js

```diff
@@ -300,6 +332,24 @@ function process_notification(notification) {
         content = message.sender_full_name + content.slice(3);
     }
 
+    // We provide desktop app with tag (message id) which is also
+    // passed in to Notification so desktop app can match the data sent with the
+    // notification. When it recives a inline reply from the notification
+    // it then sends back the `type`, `to`, and `topic` along `content` the user typed to
+    // 'send_inline_notification_message' event which we use to send a message.
+    // Desktop app cannot send a request themselves since they need to depend on
+    // channel.post to do set csrf_token so we do that on webapp side of code instead
+    // in the send_inline_notification_message event above; it also allows them to not
+    // depend on narrow.by_topic.
+    if (window.electron_bridge !== undefined) {
+        window.electron_bridge.send_event('inline_notification_data', {
+            tag: message.id,
+            type: message.type,
+            to: message.type === 'private' ? message.reply_to : message.stream,
+            topic: topic,
+        });
+    }
```

Comment:
Would it make sense to send these data via the `notification_object` below, instead?  

I don't know precisely how this fits together, but I think that'd be a much cleaner model (to have a conditional that extends what we include in `notification_object`), both in terms of this code reading linearly and in the terms of the electron app code not having to stitch together data about a notification from two sources.

Reply:
That sounds good too. I moved it send the data via `notifcation_object`.

Reply:
I also updated PR description with updated test code for testing this in desktop app.